contribs/gnodev/cmd/gnodev/setup_node.go

```diff
@@ -10,43 +10,57 @@ import (
 	gnodev "github.com/gnolang/gno/contribs/gnodev/pkg/dev"
 	"github.com/gnolang/gno/contribs/gnodev/pkg/emitter"
 	"github.com/gnolang/gno/gno.land/pkg/gnoland"
-	"github.com/gnolang/gno/tm2/pkg/std"
+	"github.com/gnolang/gno/tm2/pkg/bft/types"
 )
 
 // setupDevNode initializes and returns a new DevNode.
 func setupDevNode(
 	ctx context.Context,
-	logger *slog.Logger,
-	cfg *devCfg,
-	remitter emitter.Emitter,
-	balances gnoland.Balances,
-	pkgspath []gnodev.PackagePath,
+	devCfg *devCfg,
+	nodeConfig *gnodev.NodeConfig,
 ) (*gnodev.Node, error) {
-	// Load transactions.
-	txs, err := parseTxs(cfg.txsFile)
-	if err != nil {
-		return nil, fmt.Errorf("unable to load transactions: %w", err)
+	logger := nodeConfig.Logger
+
+	if devCfg.txsFile != "" { // Load txs files
+		var err error
+		nodeConfig.InitialTxs, err = parseTxs(devCfg.txsFile)
+		if err != nil {
+			return nil, fmt.Errorf("unable to load transactions: %w", err)
+		}
+	} else if devCfg.genesisFile != "" { // Load genesis file
+		state, err := extractAppStateFromGenesisFile(devCfg.genesisFile)
+		if err != nil {
+			return nil, fmt.Errorf("unable to load genesis file %q: %w", devCfg.genesisFile, err)
+		}
+
+		// Override balances and txs
+		nodeConfig.BalancesList = state.Balances
```

Comment:
Where do we handle the balances file? (update the nodeConfig.BalancesList based on the passed in balances sheet)

Reply:
in the method bellow `setupDevNodeConfig`, I think it's fine for now, you cannot specify `genesis-file` with `balance-file`